File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.go
func largestInteger(nums []int, k int) int {
    freq := make([]int, 51)
    for i := 0; i <= len(nums) - k; i++ {
        seen := map[int]bool{}
        for j := i; j < i + k; j++ {
            if !seen[nums[j]] {
                seen[nums[j]] = true
                freq[nums[j]]++
            }
        }
    }

    ans := -1
    for num, f := range freq {
        if f == 1 {
            ans = max(ans, num)
        }
    }

    return ans
}

// array, hash map
// time: O(nk)
// space: O(k)

File: 3471-find-the-largest-almost-missing-integer/3471-find-the-largest-almost-missing-integer.py
class Solution:
    def largestInteger(self, nums: List[int], k: int) -> int:
        freq = [0] * 51
        for i in range(len(nums) - k + 1):
            seen = set()
            for j in range(i, i + k):
                if nums[j] not in seen:
                    seen.add(nums[j])
                    freq[nums[j]] += 1
        
        ans = -1
        for num in range(len(freq)):
            if freq[num] == 1:
                ans = max(ans, num)
        
        return ans

# array, hash map
# time: O(nk)
# space: O(k)